components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1863,4 +1865,32 @@ private void validateScopeExistence(String scopeName) throws IdentityOAuthAdminE
                             scopeName));
         }
     }
+
+    /**
+     * Returns OIDC scopes registered in the tenant.
+     *
+     * @param tenantName tenant name
+     * @return List of OIDC scopes registered in tenant.
+     * @throws IdentityOAuthAdminException exception if OIDC scope retrieval fails.
+     */
+    public List<String> getRegisteredOIDCScope(String tenantName) throws IdentityOAuthAdminException {
+
+
+        try {
+            int tenantId = getTenantId(tenantName);
+            List<String> oidcScopes = OAuthTokenPersistenceFactory.getInstance().getScopeClaimMappingDAO().
+                    getScopeNames(tenantId);
+
+            if (CollectionUtils.isEmpty(oidcScopes)) {
```

Comment:
Is there a possibility for DAO to return null?
if not we can redirectly return the result from DAO